src/datasets/metric.py

```diff
@@ -336,7 +336,8 @@ def _finalize(self):
         from all the nodes if main node or all_process is True.
         """
         if self.writer is not None:
-            self.writer.finalize()
+            with self.writer as writer:
+                writer.finalize()
         self.writer = None
```

Comment:
Hi @lhoestq , do you think this is the cause of the test error? I am trying to understand why... It happens with `DistributedDatasetTest.test_load_dataset_distributed`. Maybe you know why...